src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -598,9 +604,26 @@ private unsafe uint HandleEventRecv(ref MsQuicNativeMethods.StreamEvent evt)
             return MsQuicStatusCodes.Pending;
         }
 
-        private uint HandleEventPeerRecvAbort()
+        private uint HandleEventPeerRecvAborted(ref StreamEvent evt)
         {
             if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+
+            bool shouldComplete = false;
+            lock (_sync)
+            {
+                if (_sendState == SendState.None)
+                {
+                    shouldComplete = true;
+                }
+                _sendState = SendState.Aborted;
+                _sendErrorCode = evt.Data.PeerSendAbort.ErrorCode;
```

Comment:
Make sendErrorCode a local variable. I don't believe it is used elsewhere.

Reply:
It is used to throw a `QuicStreamAbortedException` on the other side of the `ValueTask`